### ISSUE TITLE
Convolution fails when kernel sums to zero

### ISSUE THREAD

**Issue by larrybradley** [astropy/astropy#1647]
The convolve (convolution.convolve) function always normalizes the kernel by its sum prior to performing the convolution:

```
    # Because the Cython routines have to normalize the kernel on the fly, we
    # explicitly normalize the kernel here, and then scale the image at the
    # end if normalization was not requested.
    kernel_sum = kernel_internal.sum()
    kernel_internal /= kernel_sum
```

and scales the result after the convolution:

```
    # If normalization was not requested, we need to scale the array (since
    # the kernel was normalized prior to convolution)
    if not normalize_kernel:
        result *= kernel_sum
```

Obviously, this procedure fails when kernel_sum = 0.  This can occur when using sharpening or edge-detection filters (e.g. kern = [[0, -1, 0], [-1, 4, -1], [0, -1, 0]]).  Scipy's ndimage.convolve handles such a zero-sum kernel just fine.

**Comment by cdeil** [comment]
I haven't imported `astropy.convolve` in a long time and am busy with other things, so please un-assign me.

**Comment by keflavich** [comment]
Closed by #5782
